### ISSUE TITLE
0.9.0-beta2 circular reference error

### ISSUE THREAD

**Issue by gerwitz** [11ty/eleventy#660]
0.9.0-beta introduces a regression in the behavior of `eleventyExcludeFromCollections`. In my instance, I trigger:
```
PaginationError: Pagination circular reference, data:`collections.microblog` iterates over both the `microblog` tag and also supplies pages to that tag.
```
This is caused by an `index.njk` that sits aside a directory data file that adds the `microblog` tag, while the file itself does the pagination thing with it:
```yaml
pagination:
  data: collections.microblog
```
With 0.8.3 this can be resolved with `eleventyExcludeFromCollections: true`. But with 0.9.0-beta2 that does not resolve this error and I have to instead explicitly clear the tag on `index.njk` thusly:
```yaml
tags:
```
and disable `setDataDeepMerge`.

(I did not test 0.9.0-beta1. #513 sure would have made this easier to diagnose!)

**Comment by zachleat** [comment]
A ha, yes—this makes sense in the code. Will fix for 0.9.0-beta.3

**Comment by zachleat** [comment]
This code is good to go and will ship with 0.9.0-beta.3

Thank you so much for beta testing!
